server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -341,10 +348,17 @@ private boolean upload(long primaryTerm, long generation, long maxSeqNo) throws
                 transferSnapshotProvider,
                 new RemoteFsTranslogTransferListener(generation, primaryTerm, maxSeqNo)
             );
+        } finally {
+            syncPermit.release(SYNC_PERMIT);
         }
 
     }
 
+    // Made available for testing only
+    void induceDelayForTest() {
```

Comment:
merely for testing a case which otherwise is difficult to test.

Reply:
The way to do this is to use an overridden(mock) version of `RemoteFsTranslog` that induces the delay before the base method invocation